pandas-stubs/core/groupby/generic.pyi

```diff
@@ -1,150 +1,191 @@
 from pandas.core.frame import DataFrame
-from pandas.core.generic import NDFrame
-from pandas.core.groupby.groupby import (  # , get_groupby as get_groupby
-    GroupBy as GroupBy,
+from pandas.core.groupby.groupby import (
+    GroupBy,
+    GroupByPlot,
 )
-from pandas.core.groupby.grouper import Grouper
 from pandas.core.series import Series
-from typing_extensions import TypeAlias
+from typing_extensions import (
+    Self,
+    TypeAlias,
+)
 
+from pandas._libs.lib import NoDefault
 from pandas._typing import (
     S1,
     AggFuncTypeBase,
     AggFuncTypeFrame,
+    ArrayLike,
     Axis,
     ByT,
+    CorrelationMethod,
+    Dtype,
+    FillnaOptions,
+    IndexLabel,
     Level,
     ListLike,
-    RandomState,
     Scalar,
+    TakeIndexer,
+    WindowingEngine,
+    WindowingEngineKwargs,
 )
 
+from pandas.plotting import boxplot_frame_groupby
+
 AggScalar: TypeAlias = str | Callable[..., Any]
-ScalarResult = ...
+ScalarResult = TypeVar("ScalarResult")  # noqa: PYI001
 
 class NamedAgg(NamedTuple):
-    column: str = ...
-    aggfunc: AggScalar = ...
-
-def generate_property(name: str, klass: type[NDFrame]): ...
+    column: str
+    aggfunc: AggScalar
 
-class SeriesGroupBy(GroupBy, Generic[S1, ByT]):
-    def any(self, skipna: bool = ...) -> Series[bool]: ...
-    def all(self, skipna: bool = ...) -> Series[bool]: ...
-    def apply(self, func, *args, **kwargs) -> Series: ...
+class SeriesGroupBy(GroupBy[Series[S1]], Generic[S1, ByT]):
     @overload
-    def aggregate(self, func: list[AggFuncTypeBase], *args, **kwargs) -> DataFrame: ...
+    def aggregate(
+        self,
+        func: list[AggFuncTypeBase],
+        *args,
+        engine: WindowingEngine = None,
+        engine_kwargs: WindowingEngineKwargs = None,
+        **kwargs,
+    ) -> DataFrame: ...
     @overload
-    def aggregate(self, func: AggFuncTypeBase, *args, **kwargs) -> Series: ...
+    def aggregate(
+        self,
+        func: AggFuncTypeBase | None = None,
+        *args,
+        engine: WindowingEngine = None,
+        engine_kwargs: WindowingEngineKwargs = None,
+        **kwargs,
+    ) -> Series: ...
     agg = aggregate
-    def transform(self, func: Callable | str, *args, **kwargs) -> Series: ...
-    def filter(self, func, dropna: bool = ..., *args, **kwargs): ...
-    def nunique(self, dropna: bool = ...) -> Series: ...
-    def describe(self, **kwargs) -> DataFrame: ...
+    def transform(
+        self,
+        func: Callable | str,
+        *args,
+        engine: WindowingEngine = None,
+        engine_kwargs: WindowingEngineKwargs = None,
+        **kwargs,
+    ) -> Series: ...
+    def filter(
+        self, func: Callable | str, dropna: bool = True, *args, **kwargs
+    ) -> Series: ...
+    def nunique(self, dropna: bool = True) -> Series[int]: ...
+    # describe delegates to super() method but here it has keyword-only parameters
+    def describe(  # type: ignore[override] # pyright: ignore[reportIncompatibleMethodOverride]
```

Comment:
Feel free to open  PRs at pandas for those inconsistencies (preferably smaller PRs: typically easier to review/merge).